api/src/main/java/org/openmrs/module/fgh/mpi/MpiHttpClient.java

```diff
@@ -146,20 +283,44 @@ public void submitPatient(String patientData) throws Exception {
 	 * @throws Exception
 	 */
 	private <T> T submitRequest(String requestPath, String data, Class<T> responseType) throws Exception {
-		initIfNecessary();
+		AdministrationService adminService = Context.getAdministrationService();
+		
+		serverBaseUrl = adminService.getGlobalProperty(GP_MPI_BASE_URL);
```

Comment:
`serverBaseUrl` already has a value at this point, why are you reading the GP value again here?

Reply:
Ok